contracts/OrderRFQMixin.sol

```diff
@@ -60,14 +40,48 @@ abstract contract OrderRFQMixin is EIP712, AmountCalculator {
     /// @param makingAmount Making amount
     /// @param takingAmount Taking amount
     function fillOrderRFQ(
-        OrderRFQ memory order,
+        OrderRFQLib.OrderRFQ memory order,
         bytes calldata signature,
         uint256 makingAmount,
         uint256 takingAmount
     ) external returns(uint256 /* makingAmount */, uint256 /* takingAmount */, bytes32 /* orderHash */) {
         return fillOrderRFQTo(order, signature, makingAmount, takingAmount, msg.sender);
     }
 
+    uint256 constant private _MAKER_AMOUNT_FLAG = 1 << 255;
+    uint256 constant private _SIGNER_SMART_CONTRACT_HINT = 1 << 254;
+    uint256 constant private _IS_VALID_SIGNATURE_65_BYTES = 1 << 253;
+    uint256 constant private _AMOUNT_MASK = ~uint256(
+        _MAKER_AMOUNT_FLAG |
+        _SIGNER_SMART_CONTRACT_HINT |
+        _IS_VALID_SIGNATURE_65_BYTES
+    );
+
+    function fillOrderRFQCompact(
+        OrderRFQLib.OrderRFQ memory order,
+        bytes32 r,
+        bytes32 vs,
+        uint256 amount
+    ) external returns(uint256 filledMakingAmount, uint256 filledTakingAmount, bytes32 orderHash) {
+        orderHash = _hashTypedDataV4(order.hash());
+        if (amount & _SIGNER_SMART_CONTRACT_HINT != 0) {
+            if (amount & _IS_VALID_SIGNATURE_65_BYTES != 0) {
+                require(ECDSA.isValidSignature65(order.maker, orderHash, r, vs), "LOP: bad signature");
+            } else {
+                require(ECDSA.isValidSignature(order.maker, orderHash, r, vs), "LOP: bad signature");
+            }
+        } else {
+            require(ECDSA.recoverOrIsValidSignature(order.maker, orderHash, r, vs), "LOP: bad signature");
+        }
+
+        if (amount & _MAKER_AMOUNT_FLAG != 0) {
+            (filledMakingAmount, filledTakingAmount) = _fillOrderRFQTo(order, amount & _AMOUNT_MASK, 0, msg.sender);
+        } else {
+            (filledMakingAmount, filledTakingAmount) = _fillOrderRFQTo(order, 0, amount, msg.sender);
```

Comment:
@galekseev discovered bug in this line and fixed in https://github.com/1inch/limit-order-protocol/pull/131